test/xbgpu/test_engine.py

```diff
@@ -347,41 +347,98 @@ def verify_corrprod_sensors(
     weights: np.ndarray,
     quant_gains: np.ndarray,
     delays: np.ndarray,
 ) -> None:
-    """Verify katcp sensors for BPipeline data.
+    """Verify katcp sensors and Prometheus counters for BPipeline data.
+
+    NOTE: There is currently logic in place to compensate for the BPipeline's
+    lack of tracking missing data. More specifically, it compensates for the
+    difference in expected versus measured counter values by seemingly
+    'forcing' them to match. In actuality, it's counteracting the BPipeline's
+    BSend transmitting `HEAPS_PER_FENGINE_PER_CHUNK` heaps regardless of how
+    empty/full the incoming Chunk is.
 
     Parameters
     ----------
     beam_outputs
         Output beam configurations parsed into BOutput objects.
-    first_timestamp, last_timestamp
-        Two timestamps indicating the start and end of data processing
-        by the :class:`BPipeline`.
+    beam_results_shape
+        The shape of the verified beam data for all beams with the
+        following shape
+        - (len(beam_outputs), n_beam_heaps_sent, n_channels_per_substream,
+        n_samples_between_spectra, COMPLEX)
+    prom_diff
+        Collection of Prometheus metrics observed during the XBEngine's
+        processing of data stimulus.
     actual_sensor_updates
         Dictionary of lists of sensor updates. They dictionary keys are sensor
         names, the values are a list of tuples for each sensor update captured
         via the callback attached to :class:`BPipeline` sensors. Accommodating
         for three value types as there are three different types of sensors in
         the XBEngine.
+    first_timestamp, last_timestamp
+        Two timestamps indicating the start and end of data processing
+        by the :class:`BPipeline`.
+    n_heaps_total
+        Total number of heaps transmitted by the beam data stream - including
+        descriptors.
     weights
         The beam weights applied to each input of the beam data product.
         These are real floating-point values generated for the unit test.
     delays
         The beam delays applied to each beam data product. These are
-
-    .. todo::
-
-        Add verification of Prometheus counters once NGC-1154
-        is implemented.
     """
-    # Generate expected sensor updates from BPipeline and compare
+    # Get the number of total heaps transmitted by each beam output
+    # NOTE: We round up to the nearest multiple of `HEAPS_PER_FENGINE_PER_CHUNK`
+    # as the BSend currently transmits all Frames of data, regardless of how
+    # empty/full an input (received Chunk) is. Also, the number of Frames in a
+    # BSend Chunk is the same as `HEAPS_PER_FENGINE_PER_CHUNK`.
+    n_beam_heaps_sent = beam_results_shape[1]
+    n_beam_heaps_sent += n_beam_heaps_sent % HEAPS_PER_FENGINE_PER_CHUNK
+    beam_data_shape = beam_results_shape[2:]
+    # NOTE: Explicitly cast np.prod returns to int as the default is np.int64
+    heap_bytes = int(np.prod(beam_data_shape))
```

Comment:
This assumes each element is 1 byte. Maybe pass in the dtype and multiple through by its itemsize.